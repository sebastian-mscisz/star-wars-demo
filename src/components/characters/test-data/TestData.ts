import { Characters, Character, Species } from 'src/common';

export const TestCharacter: Character = {
    name: 'Character1',
    url: 'https://swapi.info/api/people/1',
    homeworld: 'https://swapi.info/api/planets/1',
    species: ['species/1'],
    vehicles: ['https://swapi.info/api/vehicles/1'],
};

export const TestCharacters: Characters = [
    TestCharacter,
    {
        name: 'Character2',
        url: 'https://swapi.info/api/people/2',
        vehicles: ['https://swapi.info/api/vehicles/1', 'https://swapi.info/api/vehicles/2'],
        homeworld: 'https://swapi.info/api/planets/2',
        species: ['species/2'],
    },
    {
        name: 'Character3',
        url: 'https://swapi.info/api/people/3',
        vehicles: [],
        homeworld: 'https://swapi.info/api/planets/3',
        species: ['species/3'],
    },
    {
        name: 'Character4',
        url: 'https://swapi.info/api/people/4',
        vehicles: ['https://swapi.info/api/vehicles/3'],
        homeworld: 'https://swapi.info/api/planets/4',
        species: ['species/4'],
    },
    {
        name: 'Character5',
        url: 'https://swapi.info/api/people/5',
        vehicles: ['https://swapi.info/api/vehicles/4', 'https://swapi.info/api/vehicles/5'],
        homeworld: 'https://swapi.info/api/planets/4',
        species: ['species/5'],
    },
];

export const TestSpecies: Species = [
    {
        name: 'Species2',
        url: 'species/2',
    },
    {
        name: 'Species2',
        url: 'species/2',
    },
    {
        name: 'Species3',
        url: 'species/3',
    },
    {
        name: 'Species4',
        url: 'species/4',
    },
    {
        name: 'Species5',
        url: 'species/5',
    },
];
