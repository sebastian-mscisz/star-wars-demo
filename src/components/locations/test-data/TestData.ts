import { Locations, Location } from 'src/common';

export const TestLocation: Location = {
    name: 'Location1',
    url: 'https://swapi.info/api/planets/1',
    residents: ['https://swapi.info/api/people/1'],
    population: '1000',
};

export const TestLocations: Locations = [
    TestLocation,
    {
        name: 'Location2',
        url: 'https://swapi.info/api/planets/2',
        residents: ['https://swapi.info/api/people/2'],
        population: '5343',
    },
    {
        name: 'Location3',
        url: 'https://swapi.info/api/planets/3',
        residents: ['https://swapi.info/api/people/3'],
        population: '13333',
    },
    {
        name: 'Location4',
        url: 'https://swapi.info/api/planets/4',
        residents: ['https://swapi.info/api/people/4', 'https://swapi.info/api/people/5'],
        population: '9999',
    },
    {
        name: 'Location5',
        url: 'https://swapi.info/api/planets/5',
        residents: [],
        population: '999',
    },
];
