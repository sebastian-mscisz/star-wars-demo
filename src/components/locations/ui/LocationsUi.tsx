import { Locations, locationsPagePath, sortAZ } from 'src/common';
import { LocationsCard, ItemsWithPagination } from 'src/components';
import './LocationsUi.scss';
import { useParams } from 'react-router-dom';

type LocationsUiProps = {
    locations?: Locations;
};

const itemsPerPage = 12;

export const LocationsUi: React.FC<LocationsUiProps> = ({ locations }) => {
    const pageId = useParams();
    const currentPageId = Number(Object.values(pageId)[0]) || 1;

    if (!locations) return null;

    const sortedLocations = [...locations].sort((a, b) => sortAZ(a.name || '', b.name || ''));

    return (
        <ItemsWithPagination
            items={sortedLocations}
            itemsPerPage={itemsPerPage}
            currentPageId={currentPageId}
            navPagePath={locationsPagePath()}
            className={'locations'}
            CardComponent={LocationsCard}
        />
    );
};
