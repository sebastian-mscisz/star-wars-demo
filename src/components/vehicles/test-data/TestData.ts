import { Vehicles, Vehicle } from 'src/common';

export const TestVehicle: Vehicle = {
    name: 'Vehicle1',
    url: 'https://swapi.info/api/vehicles/1',
    vehicle_class: 'Air speeder',
    pilots: ['character/1', 'character/2'],
};

export const TestVehicles: Vehicles = [
    TestVehicle,
    {
        name: 'Vehicle2',
        url: 'https://swapi.info/api/vehicles/2',
        pilots: ['character/2'],
        vehicle_class: 'Land speeder',
    },
    {
        name: 'Vehicle3',
        url: 'https://swapi.info/api/vehicles/3',
        pilots: ['character/4'],
        vehicle_class: 'Water speeder',
    },
    {
        name: 'Vehicle4',
        url: 'https://swapi.info/api/vehicles/4',
        pilots: ['character/5'],
        vehicle_class: 'Fire speeder',
    },
    {
        name: 'Vehicle5',
        url: 'https://swapi.info/api/vehicles/5',
        pilots: ['character/5'],
        vehicle_class: 'Air speeder',
    },
];
