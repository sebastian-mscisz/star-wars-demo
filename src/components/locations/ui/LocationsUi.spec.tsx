import { BrowserRouter } from 'react-router-dom';
import { LocationsUi } from './LocationsUi';
import { render } from '@testing-library/react';
import { TestLocations } from '../test-data';

describe('LocationsUi', () => {
    it('should render', () => {
        const { asFragment } = render(<LocationsUi locations={TestLocations} />, { wrapper: BrowserRouter });
        expect(asFragment()).toMatchSnapshot();
    });
});
