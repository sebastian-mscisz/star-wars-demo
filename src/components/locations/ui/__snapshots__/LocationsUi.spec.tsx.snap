// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationsUi should render 1`] = `
<DocumentFragment>
  <div
    class="sw-container__rows-wrapper"
  >
    <div
      class="sw-pagination-section"
      data-testid="pagination-section"
    >
      <div
        class="sw-pagination-button sw-pagination-button--hidden"
        data-testid="left-pagination-button"
      >
        ←
      </div>
      <div
        class="sw-pagination-button sw-pagination-button--hidden"
        data-testid="right-pagination-button"
      >
        →
      </div>
    </div>
    <div
      class="sw-locations"
    >
      <div
        class="sw-locations-card"
      >
        <div
          class="sw-locations-card__tile"
        >
          <a
            class="sw-locations-card__tile__link"
            href="/location/1"
          >
            <h5
              class="sw-locations-card__tile__link__text"
            >
              Location1
            </h5>
            <div
              class="sw-locations-card__tile__link__image-wrapper"
            >
              <div
                class="sw-locations-card__tile__link__image-wrapper__display-spinner"
              >
                <div
                  class="sw-spinner"
                >
                  <span
                    style="display: inherit; position: relative; width: 25px; height: 5px; overflow: hidden; background-color: black; background-clip: padding-box; border-radius: 5px;"
                  >
                    <span
                      style="position: absolute; height: 5px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-long Infinitys  cubic-bezier(0.65, 0.815, 0.735, 0.395) infinite;"
                    />
                    <span
                      style="position: absolute; height: 5px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-short Infinitys Infinitys cubic-bezier(0.165, 0.84, 0.44, 1) infinite;"
                    />
                  </span>
                  <span
                    style="display: inherit; position: relative; width: 100px; height: 5px; overflow: hidden; background-color: rgba(0, 0, 0, 0.2); background-clip: padding-box; border-radius: 5px;"
                  >
                    <span
                      style="position: absolute; height: 5px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-long 2.1s  cubic-bezier(0.65, 0.815, 0.735, 0.395) infinite;"
                    />
                    <span
                      style="position: absolute; height: 5px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-short 2.1s 1.15s cubic-bezier(0.165, 0.84, 0.44, 1) infinite;"
                    />
                  </span>
                </div>
              </div>
              <img
                alt="Star wars planet"
                class="sw-locations-card__tile__link__image-wrapper__hide_img"
                src="./locations/1.jpg"
              />
            </div>
          </a>
        </div>
      </div>
      <div
        class="sw-locations-card"
      >
        <div
          class="sw-locations-card__tile"
        >
          <a
            class="sw-locations-card__tile__link"
            href="/location/2"
          >
            <h5
              class="sw-locations-card__tile__link__text"
            >
              Location2
            </h5>
            <div
              class="sw-locations-card__tile__link__image-wrapper"
            >
              <div
                class="sw-locations-card__tile__link__image-wrapper__display-spinner"
              >
                <div
                  class="sw-spinner"
                >
                  <span
                    style="display: inherit; position: relative; width: 25px; height: 5px; overflow: hidden; background-color: black; background-clip: padding-box; border-radius: 5px;"
                  >
                    <span
                      style="position: absolute; height: 5px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-long Infinitys  cubic-bezier(0.65, 0.815, 0.735, 0.395) infinite;"
                    />
                    <span
                      style="position: absolute; height: 5px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-short Infinitys Infinitys cubic-bezier(0.165, 0.84, 0.44, 1) infinite;"
                    />
                  </span>
                  <span
                    style="display: inherit; position: relative; width: 100px; height: 5px; overflow: hidden; background-color: rgba(0, 0, 0, 0.2); background-clip: padding-box; border-radius: 5px;"
                  >
                    <span
                      style="position: absolute; height: 5px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-long 2.1s  cubic-bezier(0.65, 0.815, 0.735, 0.395) infinite;"
                    />
                    <span
                      style="position: absolute; height: 5px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-short 2.1s 1.15s cubic-bezier(0.165, 0.84, 0.44, 1) infinite;"
                    />
                  </span>
                </div>
              </div>
              <img
                alt="Star wars planet"
                class="sw-locations-card__tile__link__image-wrapper__hide_img"
                src="./locations/2.jpg"
              />
            </div>
          </a>
        </div>
      </div>
      <div
        class="sw-locations-card"
      >
        <div
          class="sw-locations-card__tile"
        >
          <a
            class="sw-locations-card__tile__link"
            href="/location/3"
          >
            <h5
              class="sw-locations-card__tile__link__text"
            >
              Location3
            </h5>
            <div
              class="sw-locations-card__tile__link__image-wrapper"
            >
              <div
                class="sw-locations-card__tile__link__image-wrapper__display-spinner"
              >
                <div
                  class="sw-spinner"
                >
                  <span
                    style="display: inherit; position: relative; width: 25px; height: 5px; overflow: hidden; background-color: black; background-clip: padding-box; border-radius: 5px;"
                  >
                    <span
                      style="position: absolute; height: 5px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-long Infinitys  cubic-bezier(0.65, 0.815, 0.735, 0.395) infinite;"
                    />
                    <span
                      style="position: absolute; height: 5px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-short Infinitys Infinitys cubic-bezier(0.165, 0.84, 0.44, 1) infinite;"
                    />
                  </span>
                  <span
                    style="display: inherit; position: relative; width: 100px; height: 5px; overflow: hidden; background-color: rgba(0, 0, 0, 0.2); background-clip: padding-box; border-radius: 5px;"
                  >
                    <span
                      style="position: absolute; height: 5px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-long 2.1s  cubic-bezier(0.65, 0.815, 0.735, 0.395) infinite;"
                    />
                    <span
                      style="position: absolute; height: 5px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-short 2.1s 1.15s cubic-bezier(0.165, 0.84, 0.44, 1) infinite;"
                    />
                  </span>
                </div>
              </div>
              <img
                alt="Star wars planet"
                class="sw-locations-card__tile__link__image-wrapper__hide_img"
                src="./locations/3.jpg"
              />
            </div>
          </a>
        </div>
      </div>
      <div
        class="sw-locations-card"
      >
        <div
          class="sw-locations-card__tile"
        >
          <a
            class="sw-locations-card__tile__link"
            href="/location/4"
          >
            <h5
              class="sw-locations-card__tile__link__text"
            >
              Location4
            </h5>
            <div
              class="sw-locations-card__tile__link__image-wrapper"
            >
              <div
                class="sw-locations-card__tile__link__image-wrapper__display-spinner"
              >
                <div
                  class="sw-spinner"
                >
                  <span
                    style="display: inherit; position: relative; width: 25px; height: 5px; overflow: hidden; background-color: black; background-clip: padding-box; border-radius: 5px;"
                  >
                    <span
                      style="position: absolute; height: 5px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-long Infinitys  cubic-bezier(0.65, 0.815, 0.735, 0.395) infinite;"
                    />
                    <span
                      style="position: absolute; height: 5px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-short Infinitys Infinitys cubic-bezier(0.165, 0.84, 0.44, 1) infinite;"
                    />
                  </span>
                  <span
                    style="display: inherit; position: relative; width: 100px; height: 5px; overflow: hidden; background-color: rgba(0, 0, 0, 0.2); background-clip: padding-box; border-radius: 5px;"
                  >
                    <span
                      style="position: absolute; height: 5px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-long 2.1s  cubic-bezier(0.65, 0.815, 0.735, 0.395) infinite;"
                    />
                    <span
                      style="position: absolute; height: 5px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-short 2.1s 1.15s cubic-bezier(0.165, 0.84, 0.44, 1) infinite;"
                    />
                  </span>
                </div>
              </div>
              <img
                alt="Star wars planet"
                class="sw-locations-card__tile__link__image-wrapper__hide_img"
                src="./locations/4.jpg"
              />
            </div>
          </a>
        </div>
      </div>
      <div
        class="sw-locations-card"
      >
        <div
          class="sw-locations-card__tile"
        >
          <a
            class="sw-locations-card__tile__link"
            href="/location/5"
          >
            <h5
              class="sw-locations-card__tile__link__text"
            >
              Location5
            </h5>
            <div
              class="sw-locations-card__tile__link__image-wrapper"
            >
              <div
                class="sw-locations-card__tile__link__image-wrapper__display-spinner"
              >
                <div
                  class="sw-spinner"
                >
                  <span
                    style="display: inherit; position: relative; width: 25px; height: 5px; overflow: hidden; background-color: black; background-clip: padding-box; border-radius: 5px;"
                  >
                    <span
                      style="position: absolute; height: 5px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-long Infinitys  cubic-bezier(0.65, 0.815, 0.735, 0.395) infinite;"
                    />
                    <span
                      style="position: absolute; height: 5px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-short Infinitys Infinitys cubic-bezier(0.165, 0.84, 0.44, 1) infinite;"
                    />
                  </span>
                  <span
                    style="display: inherit; position: relative; width: 100px; height: 5px; overflow: hidden; background-color: rgba(0, 0, 0, 0.2); background-clip: padding-box; border-radius: 5px;"
                  >
                    <span
                      style="position: absolute; height: 5px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-long 2.1s  cubic-bezier(0.65, 0.815, 0.735, 0.395) infinite;"
                    />
                    <span
                      style="position: absolute; height: 5px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-short 2.1s 1.15s cubic-bezier(0.165, 0.84, 0.44, 1) infinite;"
                    />
                  </span>
                </div>
              </div>
              <img
                alt="Star wars planet"
                class="sw-locations-card__tile__link__image-wrapper__hide_img"
                src="./locations/5.jpg"
              />
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
